CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Add ability to set a `RemediationStrategy` for the `KubeadmControlPlane`. This can be configured by setting the `internal.advancedConfiguration.controlPlane.remidiationStrategy` value. In the future, `MachineHealthCheck.spec.maxUnhealthy` will be deprecated in favor of `RemediationStrategy`.
```

Comment:
This isn't true – the deprecation didn't happen. See [here](https://github.com/kubernetes-sigs/cluster-api/issues/10722#issuecomment-3087761842). The field has been renamed/moved into `triggerIf` to make it clearer what the behavior is. Please check the new type definition. I'm not yet familiar that much with it.

That said, I think we shouldn't offer customizing the control plane remediation strategy at all, since the defaults are totally fine. For the `MachineHealthCheck` field renames, that's a new issue, I think.

Reply:
@AndiDog - Do you think it would be best to close this PR then in favor of a new issue?